corehq/pillows/management/commands/print_elastic_mappings.py

```diff
@@ -56,4 +63,53 @@ def handle(self, cname, **options):
             mapping = fetch_elastic_mapping(index_info.alias, index_info.type)
         else:
             mapping = index_info.mapping
+        for key in iter(k.strip() for k in options["transforms"].split(",")):
+            if key:
+                self.stderr.write(f"applying transform: {key}\n", style_func=lambda x: x)
+                mapping = ALL_TRANSFORMS[key](mapping)
         pprint(mapping, namespace, stream=options["outfile"])
+
+
+def transform_multi_field(mapping, key=None):
+    """Recursively convert 'multi_field' type (deprecated since version 0.9) to
+    'string' in an Elastic mapping. The field named the same as the property is
+    no longer necessary since the top-level property becomes the default in
+    versions >=1.0.
+    See: https://www.elastic.co/guide/en/elasticsearch/reference/1.7/_multi_fields.html
+
+    Transforms:
+        "city": {
+            "fields": {
+                "city":  {"type": "string", "index": "analyzed"},
+                "exact": {"type": "string", "index": "not_analyzed"},
+            },
+            "type": "multi_field"
+        }
+    To:
+        "city": {
+            "fields": {
+                "exact": {"type": "string", "index": "not_analyzed"}
+            },
+            "type": "string"
+        }
+    """
+    if isinstance(mapping, dict):
+        if "fields" in mapping and mapping.get("type") == "multi_field":
+            mapping = mapping.copy()
+            if key is None or key not in mapping["fields"]:
+                raise ValueError(f"'multi_field' property {key!r} is missing "
+                                 f"the 'default' field: {mapping}")
+            mapping.update(mapping["fields"].pop(key))
```

Comment:
TL;DR: Yes, to the extent that it doesn't hose any of _our_ mappings.

If there were items up a level that could be clobbered by this, then that would imply that the _existing_, pre-transformed mapping was invalid (i.e. it doesn't comply with either the old v0.x syntax, or the "new" v1.x+ syntax, instead using a mix of both).

Transforms performed by this utility are safe because this tool doesn't "auto update" (as in, write) live python code. It's up to the developer to manually use this command's output to update the actual python files (copy/paste).  This manual procedure will likely never change because this utility doesn't output other objects from the mapping python modules, and it can't preserve comments. If a developer were to attempt to use this to somehow auto-update a real python module, they would also need to commit the code change (another level of safety).